gather-scheduler.js

```diff
@@ -18,99 +18,124 @@
 
 class GatherScheduler {
 
-  static _runPhase(gatherers, gatherFun) {
-    return gatherers.reduce(
-      (chain, gatherer) => chain.then(_ => gatherFun(gatherer)),
-      Promise.resolve()
-    );
+  static runSeries(fns) {
+    const args = Array.from(arguments).slice(1);
```

Comment:
I think you want to remove this, actually, since you aren't using it in the current changes